docs/aws/audit/kubernetesmonitoring/rules/ecs_cluster_container_insights_enable.mdx

```diff
@@ -23,6 +23,233 @@ CBP,GDPR,HIPAA,ISO27001
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent the misconfiguration of ECS Container Insights not being enabled in Kubernetes using AWS CLI, follow these steps:
+
+1. **Update the AWS CLI to the Latest Version:**
+   Ensure you have the latest version of the AWS CLI installed to access the most recent features and commands.
+   ```sh
+   aws --version
+   ```
+
+2. **Enable Container Insights for ECS Clusters:**
+   Use the `update-cluster-settings` command to enable Container Insights for your ECS cluster.
+   ```sh
+   aws ecs update-cluster-settings --cluster <your-cluster-name> --settings name=containerInsights,value=enabled
+   ```
+
+3. **Verify the Cluster Settings:**
+   Confirm that Container Insights has been enabled by describing the cluster settings.
+   ```sh
+   aws ecs describe-clusters --clusters <your-cluster-name> --query "clusters[0].settings"
+   ```
+
+4. **Configure CloudWatch Agent for Kubernetes:**
+   Ensure the CloudWatch agent is configured to collect metrics from your Kubernetes cluster. This involves creating a CloudWatch agent configuration file and applying it to your cluster.
+   ```sh
+   kubectl apply -f https://raw.githubusercontent.com/aws-samples/amazon-cloudwatch-container-insights/master/k8s-deployment-manifest-templates/deployment-mode/daemonset/container-insights-monitoring/quick-start/cwagent/cwagent-daemonset.yaml
+   ```
+
+By following these steps, you can ensure that ECS Container Insights are enabled for your Kubernetes clusters using the AWS CLI.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent the misconfiguration of ECS Container Insights not being enabled in Kubernetes using Python scripts, you can follow these steps:
+
+### 1. **Set Up AWS SDK for Python (Boto3)**
+First, ensure you have the AWS SDK for Python (Boto3) installed. You can install it using pip if you haven't already:
+
+```bash
+pip install boto3
+```
+
+### 2. **Create a Boto3 Session**
+Create a Boto3 session to interact with AWS services. This will allow you to authenticate and make API calls.
+
+```python
+import boto3
+
+# Create a session using your AWS credentials
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
+    region_name='YOUR_AWS_REGION'
+)
+```
+
+### 3. **Enable Container Insights for ECS Clusters**
+Use the Boto3 ECS client to enable Container Insights for your ECS clusters. This involves updating the ECS cluster settings.
+
+```python
+# Create an ECS client
+ecs_client = session.client('ecs')
+
+# List all ECS clusters
+clusters = ecs_client.list_clusters()
+
+# Enable Container Insights for each cluster
+for cluster_arn in clusters['clusterArns']:
+    ecs_client.update_cluster_settings(
+        cluster=cluster_arn,
+        settings=[
+            {
+                'name': 'containerInsights',
+                'value': 'enabled'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>